nodejs12.x/cookiecutter-typescript-app-template/{{cookiecutter.project_name}}/app/src/apps/post-app.ts

```diff
@@ -0,0 +1,44 @@
+import { ApiGatewayEvent } from '../common/apigateway/apigateway-event';
+import { ApiGatewayResponse } from '../common/apigateway/apigateway-response';
+import { TodoRepository } from '../common/todo-repository';
+import { TodoItem } from '../common/todo-item';
+
+import { LambdaApp } from './lambda-app';
+
+/**
+ * PostApp is a LambdaApp that puts a new record into DynamoDB using the API Gateway event body as the record content.
+ * 
+ */
+export class PostApp implements LambdaApp {
+    table: string;
+    repository: TodoRepository;
+    
+    constructor(table: string, repository: TodoRepository) {
+        this.table = table;
+        this.repository = repository;
+    }
+    
+    async run(event: ApiGatewayEvent): Promise<ApiGatewayResponse> {
+        
+        try {
+            const todo: TodoItem = JSON.parse(event.body);
+            
+            if (!todo.title) {
+                console.log('Body is missing the title');
+                return { statusCode: 422 };
+            } else if (!todo.isComplete) {
+                todo.isComplete = false;
+            } else if (!todo.id) {
+                console.log('Body is missing the id');
+                return { statusCode: 422 };
+            }
+            
+            await this.repository.putTodo(todo, this.table);
+            return { statusCode: 201, body: JSON.stringify(todo) };
+            
+        } catch(err) {
+            console.log(err.message);
+            return { statusCode: 500 };
```

Comment:
The json body could be in a wrong format and this it would result in a client error (400), should we have a validation section to check for shape of incoming event body, so that it matches a certain schema?

Reply:
I tweaked this to return a 400 if the body isn't valid JSON, but I feel like it's up to users to come up with a way to perform schema validation. Doesn't seem like any of the existing templates are doing any rigorous checking of the event